go/app/main.go

```diff
@@ -20,22 +24,131 @@ type Response struct {
 	Message string `json:"message"`
 }
 
+type ResponseItems struct {
+	Items []ItemInfo `json:"items"`
+}
+
+type ItemInfo struct {
+	Name          string `json:"name"`
+	Category      string `json:"category"`
+	ImageFilename string `json:"image_filename"`
+}
+
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
 func addItem(c echo.Context) error {
+	// Open and read items.json
+	var jsonItems ResponseItems
+	jsonFile, err := os.Open("items.json")
+	if err != nil {
+		fmt.Println("fail to open items.json")
```

Comment:
also return error here

Reply:
Oh I forgot to write here😂. It seems that we always need to deal with a lot of "err != nil" when using golang, so should we write a function to check for errors, or is it better to handle each error separately like my code?

Reply:
> also return error here (and in all error checks in this func)

Oh I forgot to write here😂. It seems that we always need to deal with a lot of "err != nil" when using golang, so should we write a function to check for errors, or is it better to handle each error separately like my code?